bin/supply

```diff
@@ -41,12 +40,16 @@ else
   exit 1
 fi
 
-echo "[cyberark-conjur-buildpack]: looking for cyberark-conjur in VCAP_SERVICES"
-if [[ false = $(echo $VCAP_SERVICES | jq 'has("cyberark-conjur")') ]]; then
-  echo " - No credentials for cyberark-conjur service found in VCAP_SERVICES... exit"
-  exit 1
-else
-  echo " - found it"
+# When used in CI pipelines, we want to be able to test the buildpack
+# without a Conjur Service Broker running
+if [[ "$CONJUR_BUILDPACK_BYPASS_SERVICE_CHECK" != "true" ]]; then
```

Comment:
This is one change I'm not 100% comfortable with. I want to avoid having to include a full service broker dependency for the buildpack interactions. This would slow the tests down dramatically, increase their complexity, and Conjur isn't required to make sure that the buildpack will be invoked properly by the final buildpack.

I tried faking a Conjur service instance using User Provided Services (https://cli.cloudfoundry.org/en-US/cf/create-user-provided-service.html), but I couldn't find a way to make it present in `VCAP_SERVICES` in a way that the buildpack could use that as a mock Conjur instance.

Reply:
So I hadn't thought about directly setting `VCAP_SERVICES` with `set-env`, though I'm not sure how I feel about that either.

The problem I was trying to solve was how to CI the buildpack without requiring the Conjur Service Broker to be installed. I don't actually need Conjur since I'm using the literal values in `secrets.yml`, and it adds a lot of complexity to the test to include it.

The risk with this solution is that there is now flag that could turn off the service verification in production. It's a low risk (the app developer would have to set the `CONJUR_BUILDPACK_BYPASS_SERVICE_CHECK` environment variable to "true" in their manifest or `cf push`), but I was hoping to find a way just to mock the test environment, rather than change the buildpack itself.

Reply:
Is the problem that you can't use `set-env` to insert data into `VCAP_SERVICES`? If so, we could modify the buildpack to have a configurable credential location and use `set-env` with that instead. I may not be fully grasping the root of the problem, though.